### PR TITLE
i18n improved: dotted path -> dot-notation

### PR DESCRIPTION
**Pull request by math-GH**
Ref. #5662
see discussion there https://github.com/FreshRSS/FreshRSS/pull/5662#discussion_r1568434706

Changes proposed in this pull request:

- `dotted path` -> `dot-notation`
- `dotted path` -> `dot-notated`

Pull request checklist:

- [x] clear commit messages
- [x] code manually tested

### CODE REVIEW

File: app/i18n/fr/sub.php [hunk 1]
```diff
@@ -161,8 +161,8 @@
 					'_' => 'lien (URL) de l’article',
 					'help' => 'Exemple : <code>permalink</code>',
 				),
-				'json' => 'Chemin JSON pour :',
-				'relative' => 'Chemin relatif à l’article pour :',
+				'json' => 'Dot-notation for:',	// TODO
+				'relative' => 'Dot-notated path (relative to item) for:',	// TODO
```
Comment (math-GH) [inline app/i18n/fr/sub.php]: sorry @Alkarex . Your French translation needs a rework

File: app/i18n/he/sub.php [hunk 2]
```diff
@@ -127,12 +127,12 @@
 				'xpath' => 'XPath for:',	// TODO
 			),
 			'json_dotpath' => array(
```
Comment (math-GH) [inline app/i18n/he/sub.php]: Should we rename the key `json_dotpath` too? (f.e. `json_dotnotation`?

File: app/i18n/en/sub.php [hunk 3]
```diff
@@ -127,12 +127,12 @@
 				'xpath' => 'XPath for:',
 			),
 			'json_dotpath' => array(
-				'_' => 'JSON (Dotted paths)',
+				'_' => 'JSON (dot-notation)',
```
Comment (Frenzie) [inline app/i18n/en/sub.php]:
```suggestion
				'_' => 'JSON (dot notation)',
```

File: app/i18n/en/sub.php [hunk 4]
```diff
@@ -127,12 +127,12 @@
...
 				'feed_title' => array(
 					'_' => 'feed title',
 					'help' => 'Example: <code>meta.title</code> or a static string: <code>"My custom feed"</code>',
 				),
-				'help' => 'A JSON dotted path uses dots between objects and brackets for arrays (e.g. <code>data.items[0].title</code>)',
+				'help' => 'A JSON dot-notated uses dots between objects and brackets for arrays (e.g. <code>data.items[0].title</code>)',
```
Comment (Frenzie) [inline app/i18n/en/sub.php]:
```suggestion
				'help' => 'JSON dot notation uses dots between objects and brackets for arrays (e.g. <code>data.items[0].title</code>)',
```

File: app/i18n/en/sub.php [hunk 5]
```diff
@@ -161,8 +161,8 @@
 					'_' => 'item link (URL)',
 					'help' => 'Example: <code>permalink</code>',
 				),
-				'json' => 'Dotted Path for:',
-				'relative' => 'Dotted Path (relative to item) for:',
+				'json' => 'Dot-notation for:',
```
Comment (Frenzie) [inline app/i18n/en/sub.php]:
```suggestion
				'json' => 'Dot notation for:',
```

**Review by Frenzie — COMMENTED** [review]
I'd keep them as is typical in English, meaning dot notation, not dot-notation or dotnotation.

**Comment by math-GH** [comment]
@Frenzie I would agree that `dot notation` is better than `dot-notation` but I checked via Google the term and `dot-notation` is used very often.
IMHO: I do not care which one is "better".

**Comment by Frenzie** [comment]
> I checked via Google the term and `dot-notation` is used very often.

Note that the ESLint configuration for example uses dot-notation as opposed to dot_notation or dotNotation but all of those refer to "dot notation".

File: app/i18n/he/sub.php (hunk 2 above)
Comment (Alkarex) [reply app/i18n/he/sub.php]: Yes, good idea

Comment (math-GH) [reply app/i18n/he/sub.php]: Done.
But there are much more "dotpath" attributes/variables

File: app/i18n/fr/sub.php [hunk 6]
```diff
@@ -127,12 +127,12 @@
...
 			),
 			'json_dotpath' => array(
-				'_' => 'JSON (Chemin)',
+				'_' => 'JSON (dot notation)',	// TODO
 				'feed_title' => array(
 					'_' => 'titre de flux',
 					'help' => 'Exemple : <code>meta.title</code> ou un texte statique : <code>"Mon flux personnalisé"</code>',
 				),
-				'help' => 'Un chemin JSON utilise le point comme séparateur objet, et des crochets pour un tableau : (ex : <code>data.items[0].title</code>)',
+				'help' => 'A JSON dot notated uses dots between objects and brackets for arrays (e.g. <code>data.items[0].title</code>)',	// TODO
```
Comment (Alkarex) [inline app/i18n/fr/sub.php]:
```suggestion
				'help' => 'La notation point pour JSON utilise le point comme séparateur objet, et des crochets pour un tableau : (ex : <code>data.items[0].title</code>)',
```

https://developer.mozilla.org/fr/docs/Learn/JavaScript/Objects/JSON

File: app/i18n/fr/sub.php [hunk 7]
```diff
@@ -127,12 +127,12 @@
 				'xpath' => 'XPath pour :',
 			),
 			'json_dotpath' => array(
-				'_' => 'JSON (Chemin)',
+				'_' => 'JSON (dot notation)',	// TODO
```
Comment (Alkarex) [inline app/i18n/fr/sub.php]:
```suggestion
				'_' => 'JSON (notation point)',
```

File: app/i18n/fr/sub.php [hunk 8]
```diff
@@ -161,8 +161,8 @@
 					'_' => 'lien (URL) de l’article',
 					'help' => 'Exemple : <code>permalink</code>',
 				),
-				'json' => 'Chemin JSON pour :',
-				'relative' => 'Chemin relatif à l’article pour :',
+				'json' => 'dot notation for:',	// TODO
+				'relative' => 'dot notated path (relative to item) for:',	// TODO
```
Comment (Alkarex) [inline app/i18n/fr/sub.php]:
```suggestion
				'json' => 'notation point pour :',
				'relative' => 'notation point relative à l’article pour :',
```

File: app/i18n/nl/sub.php [hunk 9]
```diff
@@ -126,13 +126,13 @@
 				'relative' => 'XPath (relatief naar bericht) voor:',
 				'xpath' => 'XPath voor:',
 			),
-			'json_dotpath' => array(
-				'_' => 'JSON (Dotted paths)',	// TODO
+			'json_dotnotation' => array(
+				'_' => 'JSON (dot notation)',	// TODO
```
Comment (Frenzie) [inline app/i18n/nl/sub.php]:
```suggestion
				'_' => 'JSON (puntnotatie)',
```

File: app/i18n/nl/sub.php [hunk 10]
```diff
@@ -126,13 +126,13 @@
...
 				'feed_title' => array(
 					'_' => 'feed title',	// TODO
 					'help' => 'Example: <code>meta.title</code> or a static string: <code>"My custom feed"</code>',	// TODO
 				),
-				'help' => 'A JSON dotted path uses dots between objects and brackets for arrays (e.g. <code>data.items[0].title</code>)',	// TODO
+				'help' => 'A JSON dot notated uses dots between objects and brackets for arrays (e.g. <code>data.items[0].title</code>)',	// TODO
```
Comment (Frenzie) [inline app/i18n/nl/sub.php]:
```suggestion
				'help' => 'JSON-puntnotatie gebruikt punten tussen objecten en vierkante haakjes voor arrays (bv. <code>data.items[0].titel</code>)',
```

File: app/i18n/nl/sub.php [hunk 11]
```diff
@@ -161,8 +161,8 @@
 					'_' => 'item link (URL)',	// TODO
 					'help' => 'Example: <code>permalink</code>',	// TODO
 				),
-				'json' => 'Dotted Path for:',	// TODO
-				'relative' => 'Dotted Path (relative to item) for:',	// TODO
+				'json' => 'dot notation for:',	// TODO
```
Comment (Frenzie) [inline app/i18n/nl/sub.php]:
```suggestion
				'json' => 'puntnotatie voor:',
```

File: app/i18n/nl/sub.php [hunk 12]
```diff
@@ -161,8 +161,8 @@
...
+				'relative' => 'dot notated path (relative to item) for:',	// TODO
```
Comment (Frenzie) [inline app/i18n/nl/sub.php]:
```suggestion
				'relative' => 'puntnotatiepad (relatief aan item) voor:',
```

**Comment by Alkarex** [comment]
I think we should update the name of the file `app/Utils/dotpathUtil.php` accordingly

**Comment by math-GH** [comment]
> I think we should update the name of the file `app/Utils/dotpathUtil.php` accordingly

could we merge this PR first and do the refactoring in another PR? It would reduce double work for the translators :)

**Comment by Frenzie** [comment]
Fine by me, and better for future git blame really, though I'm not really sure what difference it'd make to the translations?

**Comment by math-GH** [comment]
> Fine by me, and better for future git blame really, though I'm not really sure what difference it'd make to the translations?

right now they translate `dotted path` into their language and not `dot notation`. 
That is the reason why #6313 is waiting for merging this PR first

**Review by Frenzie — APPROVED** [review]

**Pull request merged**